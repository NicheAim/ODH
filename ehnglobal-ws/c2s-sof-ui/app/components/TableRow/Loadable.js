/**
 *
 * Asynchronously loads the component for TableRow
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
