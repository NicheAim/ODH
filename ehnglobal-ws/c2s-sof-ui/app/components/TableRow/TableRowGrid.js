import { Grid } from 'styled-css-grid';

const TableRowGrid = Grid.extend`
  border-bottom: 1px outset rgb(51, 51, 51);

  &:nth-child(odd) {
    background-color: #fff;
  }

  &:first-child {
    background-color: #fff;
    font-size: 20px;
  }

  &:hover {
    background-color: #dce4ef;
    cursor: pointer;
  }

  &:focus {
    background-color: #93b9ea;
  }
`;

TableRowGrid.propTypes = {};

export default TableRowGrid;
