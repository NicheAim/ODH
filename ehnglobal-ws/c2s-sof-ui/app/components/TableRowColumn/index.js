/**
 *
 * TableRowColumn
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const TableRowColumn = styled.div`
  color: #000;
  font-family: "Arial", sans-serif;
  font-style: normal;
  font-size: 12px;
  font-weight: 400;
  line-height: normal;
  text-align: left;
  margin: 0;
  word-break: break-all;
  padding: 5px;
`;

TableRowColumn.propTypes = {
  children: PropTypes.node,
};

export default TableRowColumn;
