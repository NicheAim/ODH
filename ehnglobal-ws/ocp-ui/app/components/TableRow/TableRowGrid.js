import { Grid } from 'styled-css-grid';

const TableRowGrid = Grid.extend`
  border-bottom: 1px solid #ccc;

  &:nth-child(odd) {
    background-color: #fff;
  }

  &:first-child {
    background-color: #fff;
    font-size: 20px;
  }

  &:hover {
    background-color: #fcfcfc;
    cursor: pointer;
  }

  &:focus {
    background-color: #d4efef;
  }
`;

TableRowGrid.propTypes = {};

export default TableRowGrid;
