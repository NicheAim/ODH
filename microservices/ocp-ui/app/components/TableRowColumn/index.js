/**
 *
 * TableRowColumn
 *
 */

import PropTypes from 'prop-types';
import styled from 'styled-components';

const textDecorationLineValues = ['blink', 'inherit', 'initial', 'line-through', 'none', 'overline', 'underline', 'unset'];

const TableRowColumn = styled.div`
  color: #000;
  font-family: "Arial", sans-serif;
  font-style: normal;
  font-size: 12px;
  font-weight: 400;
  line-height: 200%;
  text-align: left;
  margin: 0;
  word-break: break-word;
  padding: 5px;
  text-decoration-line: ${({ textDecorationLine }) => textDecorationLine};
`;

TableRowColumn.propTypes = {
  children: PropTypes.node,
  textDecorationLine: PropTypes.oneOf(textDecorationLineValues),
};
TableRowColumn.defaultProps = {
  textDecorationLine: 'none',
};
export default TableRowColumn;
