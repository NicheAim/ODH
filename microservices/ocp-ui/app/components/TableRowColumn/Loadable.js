/**
 *
 * Asynchronously loads the component for TableRowColumn
 *
 */

import Loadable from 'react-loadable';

export default Loadable({
  loader: () => import('./index'),
  loading: () => null,
});
