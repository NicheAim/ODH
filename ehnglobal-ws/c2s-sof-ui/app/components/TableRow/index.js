/**
 *
 * TableRow
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import pickBy from 'lodash/pickBy';

import TableRowGrid from './TableRowGrid';

function TableRow({ children, columns, onClick, onKeyPress, role, tabIndex }) {
  const additionalProps = pickBy({ onClick, onKeyPress, role, tabIndex });
  return (
    <TableRowGrid
      gap="5px"
      columns={columns || `repeat(${React.Children.count(children)}, 1fr)`}
      {...additionalProps}
    >
      {children}
    </TableRowGrid>
  );
}

TableRow.propTypes = {
  children: PropTypes.node,
  columns: PropTypes.string,
  onClick: PropTypes.func,
  onKeyPress: PropTypes.func,
  role: PropTypes.string,
  tabIndex: PropTypes.string,
};

export default TableRow;
