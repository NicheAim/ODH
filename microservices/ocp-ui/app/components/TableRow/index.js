/**
 *
 * TableRow
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import pickBy from 'lodash/pickBy';

import TableRowGrid from './TableRowGrid';

function TableRow({ children, columns, onClick, onKeyPress, role, tabIndex, idRow }) {
  const additionalProps = pickBy({ onClick, onKeyPress, role, tabIndex });
  return (
    <TableRowGrid
      gap="0px"
      columns={columns || `repeat(${React.Children.count(children)}, 1fr)`}
      id={idRow}
      {...additionalProps}
    >
      {children}
    </TableRowGrid>
  );
}

TableRow.propTypes = {
  children: PropTypes.node,
  columns: PropTypes.string,
  onClick: PropTypes.func,
  onKeyPress: PropTypes.func,
  role: PropTypes.string,
  tabIndex: PropTypes.string,
  id: PropTypes.string,
};

export default TableRow;
