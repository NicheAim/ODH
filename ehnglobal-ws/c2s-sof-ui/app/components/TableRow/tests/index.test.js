import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';

import TableRow from '../index';
import TableRowColumn from '../../TableRowColumn';

configure({ adapter: new Adapter() });

describe('<TableRow />', () => {
  it('should match snapshot', () => {
    // Act
    const renderedComponent = shallow(
      (<TableRow>
        <TableRowColumn>Row</TableRowColumn>
      </TableRow>)
    );

    // Assert
    expect(renderedComponent).toMatchSnapshot();
  });
});
