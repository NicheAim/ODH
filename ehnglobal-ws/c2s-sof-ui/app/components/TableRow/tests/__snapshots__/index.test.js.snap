// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableRow /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TableRow>
    <TableRowColumn__TableRowColumn>
      Row
    </TableRowColumn__TableRowColumn>
  </TableRow>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TableRowColumn__TableRowColumn>
        Row
      </TableRowColumn__TableRowColumn>,
      "columns": "repeat(1, 1fr)",
      "gap": "5px",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Row",
      },
      "ref": null,
      "rendered": "Row",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TableRowColumn__TableRowColumn>
          Row
        </TableRowColumn__TableRowColumn>,
        "columns": "repeat(1, 1fr)",
        "gap": "5px",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Row",
        },
        "ref": null,
        "rendered": "Row",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
